src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -991,6 +1032,11 @@ private static void GetSimplyNamedTypeFromAssembly(Assembly assm, string typeNam
             catch (FileNotFoundException) { }
             catch (FileLoadException) { }
             catch (BadImageFormatException) { }
+
+            if((object)type == null)
```

Comment:
Thanks, we saw Desktop code had this and though this might be changing logic around overriden equals - if this is equivalent let's remove it. cc: @morganbr